File: service_state/vs2008_cpp/TestService/TestStateB.cpp
#include "Common/Locker.h"
#include "Common/Service.h"
#include "TestService.h"
#include "TestStateB.h"

TestStateB::TestStateB()
    : TestState("TestStateB")
{
}

TestStateB::~TestStateB()
{
}

void TestStateB::event1(Service& sv, Event& ev)
{
	printf("TestStateB::event1()\n");
}

void TestStateB::event2(Service& sv, Event& ev)
{
	printf("TestStateB::event2()\n");
	((TestService*)&sv)->setEndEvent(); // I—¹
}

File: service_state/vs2008_csharp/ServiceState/State/TestStateB.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ServiceState.State
{
    class TestStateB : TestState
    {
        public TestStateB() : base("TestStateB")
        {
        }

        /*
        public override void EventA(common.Service service, common.MQEvent ev)
        {
            // 何もしない
        }
         */


        public override void EventB(Common.Service service, Common.Event ev)
        {
            System.Threading.Thread.Sleep(1000 * 1);
            ((Service.TestService)service).SendMessage(System.Text.Encoding.ASCII.GetBytes("EventC"));
            service.ChangeState("TestStateC");
        }

        /*
        public override void EventC(common.Service service, common.MQEvent ev)
        {
            // 何もしない
        }
         */

    }
}
